src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,120 @@
+            if (Objects.equals(currentElement.key, key)) {
+                currentElement.value = value;
+                size--;
+            } else {
+                while (currentElement.next != null) {
+                    if (Objects.equals(currentElement.key, key)) {
+                        currentElement.value = value;
+                        size--;
+                        return;
+                    }
+                    currentElement = currentElement.next;
+                }
+                currentElement.next = new Node<>(key, value);
+            }
+        } else {
+            table[numberOfBucket] = new Node<>(key, value);
+        }
     }
 
     @Override
     public V getValue(K key) {
+        if (table != null) {
+            int numberOfBucket = hash(key) % table.length;
+            Node<K, V> element = table[numberOfBucket];
+            if (Objects.equals(element.key, key)) {
+                return element.value;
+            } else {
+                Node<K, V> currentElement = element;
+                while (currentElement != null) {
+                    if (Objects.equals(currentElement.key, key)) {
+                        return currentElement.value;
+                    }
+                    currentElement = currentElement.next;
+                }
+            }
+        }
         return null;
     }
 
     @Override
     public int getSize() {
-        return 0;
+        return size;
+    }
+
+    private void resize() {
+        Node<K, V> [] oldTable = table;
+        int oldCapacity = (oldTable == null) ? 0 : table.length;
+        int oldThreshold = threshold;
+        int newCapacity;
+        int newThreshold;
+        if (oldCapacity == 0) {
+            newCapacity = DEFAULT_INITIAL_CAPACITY;
+            newThreshold = (int) (DEFAULT_INITIAL_CAPACITY * DEFAULT_LOAD_FACTOR);
+            table = (Node<K, V> [])new Node[newCapacity];
+        } else {
+            newCapacity = oldCapacity << 1;
+            newThreshold = oldThreshold << 1;
+            table = (Node<K, V> [])new Node[newCapacity];
+            transfer(oldTable);
+        }
+        threshold = newThreshold;
+    }
+
+    private void transfer(Node<K, V> [] oldTable) {
+        for (int i = 0; i < oldTable.length; i++) {
+            if (oldTable[i] != null) {
+                putValue(oldTable[i].key, oldTable[i].value);
+                if (oldTable[i].next != null) {
+                    Node<K, V> currentElement = oldTable[i].next;
+                    while (currentElement != null) {
+                        putValue(currentElement.key, currentElement.value);
+                        currentElement = currentElement.next;
+                    }
+                }
+                oldTable[i] = null;
+            }
+        }
+    }
+
+    private int hash(Object key) {
+        int result = 17;
+        return (key == null) ? 0 : (Math.abs(31 * result + key.hashCode()));
+    }
```

Comment:
```suggestion
    private int hash(Object key) {
        return (key == null) ? 0 : (Math.abs( key.hashCode()));
    }
```

I don't think we need this